### DISCUSSION TITLE
Got it working!  Comments from a newbie!

### DISCUSSION THREAD

**Discussion by rmeden** [tagyoureit/nodejs-poolController#887] [Q&A]
Thank you for these excellent projects.  For years I've been annoyed by the lack of reasonable connectivity to my IntelleTouch panel and how expensive and kludgy the Pentair remote solution was.  I only recently searched TroubleFreePool (TFP) to see if there were solutions.

I've been a programmer for over 40 years, professionally working as a Linux SA for much of that time and contributed to many and lead a few projects.  I've also been dealing with RS232, HDLC, X.25 and other serial communications for decades.   This  *should* be a piece of cake for me.  That said, I'm not familiar with a lot of the "in-vogue" languages and tools.  For ages, I've done most of my stuff in PERL, PHP, bash.. yes, I'm an old fart.. if it aint broke, don't' fix it!

I like to "keep it simple".  I already have a Windows-PC that is online 24x7 for video recording (NextPVR) and serving (Plex).  I'm running a RockyLinux VM on the machine as well (TeslaMate).  My plan was to put a EW11 in my PowerCenter, and run njs-PC and njs-PCD on the PC.  I initially tried the Rocky 8 VM , but stock nodeJS versions were too old.. so decided to stick with Windows.

I found the setup very frustrating... the README seems to have old info, poor directions (at least for my use case), and the TFP instruction posts were for different setups, and didn't explain where their screen shots were coming from! (now I know it's njc-PCD).  I suspect my setup should be easier than dealing with a Raspberry Pi (at least their available now.. I waited 2 years for Pi Zero 2w)

I know all devs (myself included) hate documentation!  Also, someone who knows the product is the worst person to write the docs for a newbie anyway.  All that said, I plan on releasing a document and maybe a youtube video of a fresh install to help others.   I also plan on suggesting (or submitting) changes to the README and Wiki.

To that end, I have some questions/comments.

**INTELLICENTER USERS: Do not upgrade to 2.006.  Rollback to 1.064 to use this application.**
Do not upgrade to 2.006 of what??? nodejs-poolController is at 8.0 according to the README... is that even relevant any more? 

**This code requires a physical RS485 interface**
That's certainly no longer true... I'm using the EW11 just fine!  Code has been added to the app to support it, but there is no mention in the README or the RS485 wiki page on how to configure it!

**the README talks about the first run will create config.json and it should be manually edited**
The installation link referenced doesn't exist.  I saw the place to configure the RS485 interface, but no clue how to configure it for my EW11.   Because I was stuck here, I didn't (initially) proceed to the client.  Why install a client if you can't configure the server???  This was further complicated by TPF posts showing a config GUI, but connecting to any ports weren't returning anything.

As I found out (and of course you guys all know!) you use the *CLIENT* to configure the server.  Maybe it's just me, but I think that would be handy to mention :)

I don't mean this to come across critical... as I mentioned, the devs are the worst folks to write newbie instructions!   Now's my time to contribute! 

So, if you can correct my interpretations, I'll do pull request and make some updates!

Thanks again for a great project.

**Comment by rstrouse** [comment]
Part of the issue is that this supports many platforms.  For instance

> INTELLICENTER USERS: Do not upgrade to 2.006. Rollback to 1.064 to use this application.
Do not upgrade to 2.006 of what??? nodejs-poolController is at 8.0 according to the README... is that even relevant any more?

Your panel is an IntelliTouch not IntelliCenter.  This does require rolling back to 1.064.  Pentair has released this firmware a couple of times and pulled it back.

> This code requires a physical RS485 interface
That's certainly no longer true... I'm using the EW11 just fine! Code has been added to the app to support it, but there is no mention in the README or the RS485 wiki page on how to configure it!

The EW11 is a physical RS485 device.  Unfortunately, EW11 isn't the only networked RS485 device.  They all operate using TCP streams but their configurations are different.

> the README talks about the first run will create config.json and it should be manually edited
The installation link referenced doesn't exist. I saw the place to configure the RS485 interface, but no clue how to configure it for my EW11. Because I was stuck here, I didn't (initially) proceed to the client. Why install a client if you can't configure the server??? This was further complicated by TPF posts showing a config GUI, but connecting to any ports weren't returning anything.

Yeah this will be removed.  The adapter configuration wasn't available in earlier versions of the software and those folks that are docker peeps still edit that file to set up the configuration.  However, the dashPanel is not required to run the njsPC server but I don't think there are very many users doing that anymore since you do not need a Pentair panel to run the software anymore.

> So, if you can correct my interpretations, I'll do pull request and make some updates!

Most of the detailed configuration stuff is in the Wiki.   Would love to have some pages on EW11 configuration and any help on the wiki would be appreciated.

**Comment by rmeden** [comment]
> > INTELLICENTER USERS: Do not upgrade to 2.006. Rollback to 1.064 to use this application.
> > Do not upgrade to 2.006 of what??? nodejs-poolController is at 8.0 according to the README... is that even relevant any more?

I see how it reads now...  I suggest the following...
INTELLICENTER USERS: Do not upgrade to INTELLICENTER 2.006. Rollback to 1.064 to use this application.

> Most of the detailed configuration stuff is in the Wiki. Would love to have some pages on EW11 configuration and any help on the wiki would be appreciated.

Will do.!

**Reply by tagyoureit** [reply]
Updated the readme language for Intellicenter per above suggestion.
